File: scenarios/key-vault-python-crud-secrets/golden/key_vault_crud.py
from __future__ import annotations

import os
import sys

from azure.core.exceptions import ResourceNotFoundError
from azure.identity import DefaultAzureCredential
from azure.keyvault.secrets import SecretClient


SECRET_NAME = "my-secret"
INITIAL_VALUE = "my-secret-value"
UPDATED_VALUE = "updated-value"


def run() -> None:
    vault_url = os.environ["AZURE_KEY_VAULT_URL"]
    with DefaultAzureCredential() as credential:
        with SecretClient(vault_url=vault_url, credential=credential) as client:
            try:
                client.set_secret(SECRET_NAME, INITIAL_VALUE)
                retrieved = client.get_secret(SECRET_NAME)
                print(retrieved.value)
                client.set_secret(SECRET_NAME, UPDATED_VALUE)
                deletion = client.begin_delete_secret(SECRET_NAME)
                deletion.wait()
                client.purge_deleted_secret(SECRET_NAME)
            except ResourceNotFoundError as error:
                print(f"Key Vault secret operation failed: {error}", file=sys.stderr)
                raise


if __name__ == "__main__":
    run()

File: scenarios/key-vault-typescript-crud-secrets/fixtures/baseline-33374429826/key-vault-crud.ts
import { DefaultAzureCredential } from "@azure/identity";
import { SecretClient } from "@azure/keyvault-secrets";

const secretName = "my-secret";

async function main(): Promise<void> {
  const vaultUrl = process.env.KEY_VAULT_URL;
  if (!vaultUrl) {
    throw new Error(
      "KEY_VAULT_URL is required (for example, https://<vault-name>.vault.azure.net).",
    );
  }

  const credential = new DefaultAzureCredential();
  const client = new SecretClient(vaultUrl, credential);

  await client.setSecret(secretName, "my-secret-value");
  console.log(`Created secret "${secretName}".`);

  const secret = await client.getSecret(secretName);
  if (secret.value === undefined) {
    throw new Error(`Secret "${secretName}" was read without a value.`);
  }
  console.log(`Secret value: ${secret.value}`);

  await client.setSecret(secretName, "updated-value");
  console.log(`Updated secret "${secretName}".`);

  const deletePoller = await client.beginDeleteSecret(secretName);
  await deletePoller.pollUntilDone();
  console.log(`Deleted secret "${secretName}".`);

  await client.purgeDeletedSecret(secretName);
  console.log(`Purged secret "${secretName}".`);
}

main().catch((error: unknown) => {
  const message = error instanceof Error ? error.message : String(error);
  console.error(`Key Vault operation failed: ${message}`);
  process.exitCode = 1;
});
